src/main.cc

```diff
@@ -240,6 +248,15 @@ signal_reset(void)
 	sigfillset(&sigset);
 	if (sigprocmask(SIG_UNBLOCK, &sigset, NULL) == -1)
 		say_syserror("sigprocmask");
+
+	/*
+	 * Block SIGURG on non-main threads so that
+	 * it is delivered only to the main one.
+	 */
+	sigemptyset(&sigset);
+	sigaddset(&sigset, SIGURG);
+	if (pthread_sigmask(SIG_BLOCK, &sigset, NULL) == -1)
+		say_syserror("sigprocmask");
```

Comment:
AFAICS this function (`signal_reset`) is called at fork, not on thread creation. This should probably be done somewhere in `cord_create`. Maybe, move `sigaction(SIGURG)` to the fiber module as well (e.g. to `fiber_init`) for the sake of consistency?

Reply:
`cord_start` already uses `tt_pthread_create` which block all the signals on child thread.